src/components/GalleryBrick/index.js

```diff
@@ -0,0 +1,113 @@
+import React, { Component } from 'react';
+import FontAwesome from 'react-fontawesome';
+import styles from './styles';
+
+class GalleryBrick extends Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      pages: this.props.pages || 1,
+      currentPage: this.props.currentPage || 1,
+      pageImages : Math.round(this.props.images.length / this.props.pages)
+    }
+  }
+
+  renderImages(src, s, key){
+    return (
+      <div style={ s.imgGallery }>
+        <img key={ key } style={ s.image} src={ src } />
+      </div>
+      )
+  }
+
+  renderPagesImages(images, s){
+    if (!images) return null;
+    let boxImages = [], max = (this.state.currentPage * this.state.pageImages) - this.state.pageImages;
```

Comment:
- use the "slice" method of the items array instead of manually pushing items into a new one
- this.state.pageImages should be this.props.itemsPerPage
- use a different line of code for each variable initialization